File: app/src/main/java/catchla/yep/loader/FriendshipsLoader.java
package catchla.yep.loader;

import android.accounts.Account;
import android.content.Context;

import catchla.yep.model.Friendship;
import catchla.yep.model.Paging;
import catchla.yep.provider.YepDataStore.Friendships;

/**
 * Created by mariotaku on 15/5/27.
 */
public class FriendshipsLoader extends ObjectCursorLoader<Friendship> {

    private final Account mAccount;
    private final Paging mPaging;

    public FriendshipsLoader(Context context, Account account, Paging paging) {
        super(context, Friendship.Indices.class, Friendships.CONTENT_URI, Friendships.COLUMNS, null,
                null, null);
        mAccount = account;
        mPaging = paging;
    }

}

File: src/main/kotlin/catchla/yep/loader/FriendshipsLoader.kt
package catchla.yep.loader

import android.accounts.Account
import android.content.Context
import catchla.yep.model.Friendship
import catchla.yep.model.FriendshipCursorIndices
import catchla.yep.model.FriendshipTableInfo
import catchla.yep.provider.YepDataStore.Friendships
import catchla.yep.util.Utils
import org.mariotaku.sqliteqb.library.Expression

/**
 * Created by mariotaku on 15/5/27.
 */
class FriendshipsLoader(
        context: Context,
        account: Account
) : ObjectCursorLoader<Friendship>(context, FriendshipCursorIndices::class.java) {

    init {
        uri = Friendships.CONTENT_URI
        projection = FriendshipTableInfo.COLUMNS
        selection = Expression.equalsArgs(Friendships.ACCOUNT_ID).sql
        selectionArgs = arrayOf(Utils.getAccountId(context, account))
    }

}
